src/status_im2/contexts/onboarding/create_password/view.cljs

```diff
@@ -1,42 +1,197 @@
+        {:style (style/label-color status)
+         :size  :paragraph-2}
+        text]])]])
+
+(defn password-inputs
+  [{:keys [passwords-match? on-change-password on-change-repeat-password on-input-focus
+           password-long-enough? empty-password? same-password-lenght?]}]
+  (reagent/with-let [show-hint-2? (reagent/atom false)]
+    (let [_ (when same-password-lenght? (reset! show-hint-2? true))
+          hint-1-status (if password-long-enough? :success :neutral)
+          hint-2-status (if passwords-match? :success :danger)
+          hint-2-text   (if passwords-match?
+                          (i18n/label :t/password-creation-match)
+                          (i18n/label :t/password-creation-dont-match))
+          error?        (and @show-hint-2?
+                             (not passwords-match?)
+                             (not empty-password?))]
+      [rn/view
+       [password-with-hint
+        {:hint           {:text   (i18n/label :t/password-creation-hint)
+                          :status hint-1-status
+                          :shown  true}
+         :placeholder    (i18n/label :t/password-creation-placeholder-1)
+         :on-change-text on-change-password
+         :on-focus       #(on-input-focus :password)}]
+       [rn/view {:style style/space-between-inputs}]
+       [password-with-hint
+        {:hint           {:text   hint-2-text
+                          :status hint-2-status
+                          :shown  (and (not empty-password?)
+                                       @show-hint-2?)}
+         :error?         error?
+         :placeholder    (i18n/label :t/password-creation-placeholder-2)
+         :on-change-text on-change-repeat-password
+         :on-focus       #(on-input-focus :repeat-password)
+         :on-blur        #(if empty-password?
+                            (reset! show-hint-2? false)
+                            (reset! show-hint-2? true))}]])))
+
+(def strength-status
+  {1 :very-weak
+   2 :weak
+   3 :okay
+   4 :strong
+   5 :very-strong})
+
+(defn help
+  [{{:keys [lower-case? upper-case? numbers? symbols?]} :validations
+    password-strength                                   :password-strength}]
+  [rn/view
+   [quo/strength-divider {:type (strength-status password-strength :info)}
+    (i18n/label :t/password-creation-tips-title)]
+   [rn/view {:style style/password-tips}
+    [quo/tips {:completed? lower-case?}
+     (i18n/label :t/password-creation-tips-1)]
+    [quo/tips {:completed? upper-case?}
+     (i18n/label :t/password-creation-tips-2)]
+    [quo/tips {:completed? numbers?}
+     (i18n/label :t/password-creation-tips-3)]
+    [quo/tips {:completed? symbols?}
+     (i18n/label :t/password-creation-tips-4)]]])
+
+(defn has-lower-case [s] (re-find #"[a-z]" s))
+(defn has-upper-case [s] (re-find #"[A-Z]" s))
+(defn has-numbers [s] (re-find #"\d" s))
+(defn has-symbols [s] (re-find #"[^a-zA-Z0-9\s]" s))
+(defn at-least-10-chars? [s] (>= (count s) 10))
```

Comment:
Maybe we could move these functions to `utils.string` namespace.

I would also refactor `at-least-10-chars?` function to receive a `min-chars` parameter, and pass 10 in this case.

Reply:
IMO, it makes sense to move  functions to a more general scope if we are using (or will use) them in different places.

and about `at-least-10-chars?`'s suggestion, IDK if it's worth making it more general unless we know we'll need it.

I have little knowledge of the multiple app's flows, so please help me clear this up :sweat_smile: 
Do you think we will use these functions in a different component?

Reply:
Well, we have some similar functions on `status-im.ethereum.mnemonic` which is a namespace related to wallet / seed phrases, so probably would be better to move those kind of functions to a common place. Probably we will need some when we implement wallet stuff, and also I am not sure about Reset Password flow (which is Post-MVP as far as I see on Figma designs).

Reply:
OK! Will do it

Reply:
@briansztamfater it's done!